File: platforms/kotlin/examples/core/renderer/read_texture.kt
import org.fragmentcolor.*
val renderer = Renderer()
val texture = renderer.createStorageTexture(Size(width=64u, height=64u, depth=null), TextureFormat.RGBA, null, null)
texture.write(ByteArray(64 * 64 * 4))

val bytes = renderer.readTexture(texture.id())

File: platforms/web/examples/core/renderer/read_texture.js
import { Renderer, TextureFormat } from "fragmentcolor";
const renderer = new Renderer();
const texture = await renderer.createStorageTexture([64, 64], TextureFormat.Rgba);
texture.write(new Uint8Array(64 * 64 * 4));

const bytes = await renderer.readTexture(texture.id());